sycl/source/detail/scheduler/commands.cpp

```diff
@@ -3237,8 +3235,7 @@ ur_result_t ExecCGCommand::enqueueImpQueue() {
       return AllocaCmd ? AllocaCmd->getMemAllocation() : nullptr;
     };
 
-    const std::shared_ptr<detail::kernel_impl> &SyclKernel =
-        ExecKernel->MSyclKernel;
+    const detail::kernel_impl *SyclKernel = ExecKernel->MSyclKernel.get();
```

Comment:
This isn't implied by the PR's title ;)

Reply:
Ok, I changed the titile. But if you want I can split this PR into two separate PRs.